src/Microsoft.AspNet.SignalR.Client/Connection.cs

```diff
@@ -137,10 +137,11 @@ public class Connection : IConnection, IDisposable
         /// </summary>
         /// <param name="url">The url to connect to.</param>
         public Connection(string url)
-            : this(url, (string)null)
+            :this(url, GetQueryStringFromUrl(ref url))
```

Comment:
I am not huge fan of this `ref` - Can you have two methods one that extracts the part before '?' and the other one that returns query string? This is not a perf sensitive code so it should be fine to scan the string twice.